### DISCUSSION TITLE
How to determine which nodes have changed since the prior layout

### DISCUSSION THREAD

**Discussion by ncallaway** [DioxusLabs/taffy#667] [Q&A]
When using the High Level API to (re)compute a layout using [`TaffyTree#compute_layout`](https://docs.rs/taffy/latest/taffy/tree/struct.TaffyTree.html#method.compute_layout), is there any way to get which layouts have changed from the previous iteration?

I'm basically trying to keep my renderer in sync with my taffy layout, while keeping updates as minimal as possible. Instead of re-computing my entire render state each from from the TaffyTree de novo, I'd like to update only the values that have changed.

I know that ultimately the compute_layout logic flows through [`compute_child_layout`](https://github.com/DioxusLabs/taffy/blob/e363fc8065ce370dba9fcacd1bd6575e65657c74/src/tree/taffy_tree.rs#L286), which applies caching logic on a per-node basis using `compute_cached_layout`. It'd be enough for me to know which nodes needed to be re-computed due to a cache miss (even if, after the cache miss they happen to compute to the same values they had before).

I don't see anything obvious in [`compute_cached_layout`](https://github.com/DioxusLabs/taffy/blob/e363fc8065ce370dba9fcacd1bd6575e65657c74/src/compute/mod.rs#L95) that makes me think something from the cache is exposed to me, but I'm just wondering if I've missed something anywhere that would help me efficiently figure out the changes from the prior layout.

**Comment by wilgaboury** [comment]
I've been looking a bit through the source code and was wondering the same thing. For reference, Yoga uses a `HasNewLayout` flag that makes it fairly easy to tell which parts of the tree have had layout changes. See https://www.yogalayout.dev/docs/advanced/incremental-layout.
